File: llama3/sistemaTipos/Tipos.java
package llama3.sistemaTipos;
public class Tipos {
    public static void main(String[] args) {
        double price = 50;
        String discount = "10";
        
        double finalPrice = applyDiscount(price, discount);
        System.out.println("Precio final: " + finalPrice);
    }

    public static double applyDiscount(double price, String discount) {
        // Convertir el descuento a un valor numérico (porcentaje)
        int percentDiscount = Integer.parseInt(discount.replace("%", ""));
        
        // Calcular el precio final aplicando el descuento
        double finalPrice = price - (price * (percentDiscount / 100.0));
        
        return finalPrice;
    }
}

File: llama3/sistemaTipos/tipos.rs
fn calculate_price_with_discount(price: f64, discount: &str) -> f64 {
    let percentage: f64 = discount.parse().unwrap();
    let discount_amount = price * (percentage as f64 / 100.0);
    price - discount_amount
}

fn main() {
    let price = 50.0;
    let discount = "10";
    let final_price = calculate_price_with_discount(price, discount);
    println!("El precio final es: {}", final_price);
}